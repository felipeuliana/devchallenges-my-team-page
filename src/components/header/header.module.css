:root {
  --header-font-size: 0.875rem;
  --header-spacing: 0.625rem;
  --heading-font-size: 2.25rem;
  --heading-line-size: 1.333;
  --heading-spacing: 2.375rem;
  --subtitle-font-size: 1.125rem;
}

.root {
  display: flex;
  flex-direction: column;
}

.description {
  font-size: var(--header-font-size);
  margin: 0 0 var(--header-spacing);
}

.heading {
  font-size: var(--heading-font-size);
  line-height: var(--heading-line-size);
  margin: var(--heading-spacing) 0;
}

.subtitle {
  font-size: var(--subtitle-font-size);
  margin: 0 0 var(--header-spacing);
}

@media screen and (min-width: 768px) {
  :root {
    --header-font-size: 1.125rem;
    --inner-spacing: 4.8125rem;
  }

  .root {
    flex-direction: row;
  }

  .heading {
    margin: 0;
  }

  .inner {
    flex-basis: 50%;
    padding: var(--inner-spacing) 0;
  }
}
