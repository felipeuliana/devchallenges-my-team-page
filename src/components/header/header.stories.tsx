import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import Header from './header'

export default {
  args: {
    description: 'We are team of creatively diverse.  driven.  innovative individuals working in various loctions from the world.',
    heading: 'The creative crew',
    subtitle: 'Who we are',
  },
  component: Header,
  title: 'Header',
} as ComponentMeta<typeof Header>

export const Default: ComponentStory<typeof Header> = (args) => (
  <Header {...args} />
)
