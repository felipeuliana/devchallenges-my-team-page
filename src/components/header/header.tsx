import React, { FC } from 'react'
import styles from './header.module.css'

export interface HeaderProps {
  description: string
  heading: string
  subtitle: string
}

const Header: FC<HeaderProps> = ({ description, heading, subtitle }) => (
  <header className={styles.root} data-testid="header">
    <div className={styles.inner}>
      <h1 className={styles.heading}>{heading}</h1>
    </div>

    <div className={styles.inner}>
      <h2 className={styles.subtitle}>{subtitle}</h2>
      <p className={styles.description}>{description}</p>
    </div>
  </header>
)

export default Header
