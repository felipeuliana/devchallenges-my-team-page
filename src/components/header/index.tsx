import React from 'react'
import { FC } from 'react'

interface HeaderProps {
  heading: string
  subtitle: string
}

const Header: FC<HeaderProps> = ({ heading, subtitle }) => (
  <header>
    <div>
      <h1>{heading}</h1>
    </div>

    <div>
      <h2>{subtitle}</h2>
    </div>
  </header>
)

export default Header
