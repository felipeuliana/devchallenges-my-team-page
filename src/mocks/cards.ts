export const CARDS = [
  {
    image: {
      alt: 'Avatar of Terry Medhurst',
      src: 'https://robohash.org/hicveldicta.png'
    },
    name: 'Terry Medhurst',
    title: 'Product owner'
  },
  {
    image: {
      alt: 'Avatar of Sheldon Quigley',
      src: 'https://robohash.org/doloremquesintcorrupti.png'
    },
    name: 'Sheldon Quigley',
    title: 'Art director'
  },
  {
    image: {
      alt: 'Avatar of Terrill Hills',
      src: 'https://robohash.org/consequunturautconsequatur.png'
    },
    name: 'Terrill Hills',
    title: 'Tech Lead'
  },
  {
    image: {
      alt: 'Avatar of Miles Cummerata',
      src: 'https://robohash.org/facilisdignissimosdolore.png'
    },
    name: 'Miles Cummerata',
    title: 'UX Designer'
  },
  {
    image: {
      alt: 'Avatar of Mavis Schultz',
      src: 'https://robohash.org/adverovelit.png'
    },
    name: 'Mavis Schultz',
    title: 'Developer'
  },
  {
    image: {
      alt: 'Avatar of Alison Reichert',
      src: 'https://robohash.org/laboriosamfacilisrem.png'
    },
    name: 'Alison Reichert',
    title: 'Developer'
  }
]
