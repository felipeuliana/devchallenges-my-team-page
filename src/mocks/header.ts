export const HEADER = {
  description: 'We are team of creatively diverse.  driven.  innovative individuals working in various locations from the world.',
  heading: 'The creative crew',
  subtitle: 'Who we are'
}
