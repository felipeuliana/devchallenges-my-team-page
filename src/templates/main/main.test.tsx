import React from 'react'
import { render, screen, within } from '@testing-library/react'
import Main from './main'
import { CARDS } from '../../mocks/cards'
import { HEADER } from '../../mocks/header'

describe('Main template', () => {
  let template: ChildNode | null

  beforeEach(() => {
    const { container } = render(<Main header={HEADER} cards={CARDS} />)

    template = container.firstChild
  })

  it('should render correctly', () => {
    expect(template).toMatchSnapshot()
  })

  it('should display the header', () => {
    const header = screen.getByTestId('main-header')

    expect(header).toBeInTheDocument()
  })

  it('should display the team members cards', () => {
    const cards = screen.getByTestId('main-cards')

    expect(cards).toBeInTheDocument()
    expect(within(cards).getAllByRole('listitem')).toHaveLength(6)
  })
})
