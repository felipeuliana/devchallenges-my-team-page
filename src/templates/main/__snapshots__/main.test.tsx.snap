// Vitest Snapshot v1

exports[`Main template > should render correctly 1`] = `
<main
  class="_root_41db96"
>
  <header
    class="_root_503ab9"
    data-testid="main-header"
  >
    <div
      class="_inner_503ab9"
    >
      <h1
        class="_heading_503ab9"
      >
        The creative crew
      </h1>
    </div>
    <div
      class="_inner_503ab9"
    >
      <h2
        class="_subtitle_503ab9"
      >
        Who we are
      </h2>
      <p
        class="_description_503ab9"
      >
        We are team of creatively diverse.  driven.  innovative individuals working in various locations from the world.
      </p>
    </div>
  </header>
  <ul
    class="_root_648350"
    data-testid="main-cards"
  >
    <li
      class="_root_c37972"
    >
      <figure
        class="_figure_c37972"
      >
        <img
          alt="Avatar of Terry Medhurst"
          src="https://robohash.org/hicveldicta.png"
        />
        <figcaption
          class="_caption_c37972"
        >
          Product owner
        </figcaption>
      </figure>
      <h2
        class="_name_c37972"
      >
        Terry Medhurst
      </h2>
    </li>
    <li
      class="_root_c37972"
    >
      <figure
        class="_figure_c37972"
      >
        <img
          alt="Avatar of Sheldon Quigley"
          src="https://robohash.org/doloremquesintcorrupti.png"
        />
        <figcaption
          class="_caption_c37972"
        >
          Art director
        </figcaption>
      </figure>
      <h2
        class="_name_c37972"
      >
        Sheldon Quigley
      </h2>
    </li>
    <li
      class="_root_c37972"
    >
      <figure
        class="_figure_c37972"
      >
        <img
          alt="Avatar of Terrill Hills"
          src="https://robohash.org/consequunturautconsequatur.png"
        />
        <figcaption
          class="_caption_c37972"
        >
          Tech Lead
        </figcaption>
      </figure>
      <h2
        class="_name_c37972"
      >
        Terrill Hills
      </h2>
    </li>
    <li
      class="_root_c37972"
    >
      <figure
        class="_figure_c37972"
      >
        <img
          alt="Avatar of Miles Cummerata"
          src="https://robohash.org/facilisdignissimosdolore.png"
        />
        <figcaption
          class="_caption_c37972"
        >
          UX Designer
        </figcaption>
      </figure>
      <h2
        class="_name_c37972"
      >
        Miles Cummerata
      </h2>
    </li>
    <li
      class="_root_c37972"
    >
      <figure
        class="_figure_c37972"
      >
        <img
          alt="Avatar of Mavis Schultz"
          src="https://robohash.org/adverovelit.png"
        />
        <figcaption
          class="_caption_c37972"
        >
          Developer
        </figcaption>
      </figure>
      <h2
        class="_name_c37972"
      >
        Mavis Schultz
      </h2>
    </li>
    <li
      class="_root_c37972"
    >
      <figure
        class="_figure_c37972"
      >
        <img
          alt="Avatar of Alison Reichert"
          src="https://robohash.org/laboriosamfacilisrem.png"
        />
        <figcaption
          class="_caption_c37972"
        >
          Developer
        </figcaption>
      </figure>
      <h2
        class="_name_c37972"
      >
        Alison Reichert
      </h2>
    </li>
  </ul>
</main>
`;
